mne/connectivity/tests/test_envelope.py

```diff
@@ -15,19 +15,23 @@
 def _compute_corrs_orig(data):
     # This is the version of the code by Sheraz and Denis.
     # For this version (epochs, labels, time) must be -> (labels, time, epochs)
-    data = np.transpose(data, (1, 2, 0))
-    corr_mats = np.empty((data.shape[0], data.shape[0], data.shape[2]))
-    for index, label_data in enumerate(data):
-        label_data_orth = np.imag(label_data * (data.conj() / np.abs(data)))
-        label_data_orig = np.abs(label_data)
-        label_data_cont = np.transpose(
-            np.dstack((label_data_orig, np.transpose(label_data_orth,
-                                                     (1, 2, 0)))), (1, 2, 0))
-        corr_mats[index] = np.array([np.corrcoef(dat)
-                                     for dat in label_data_cont])[:, 0, 1:].T
-    corr_mats = np.transpose(corr_mats, (2, 0, 1))
-    corr = np.mean(np.array([(np.abs(corr_mat) + np.abs(corr_mat).T) / 2.
-                             for corr_mat in corr_mats]), axis=0)
+    n_epochs, n_labels, _ = data.shape
+    corr = np.zeros((n_labels, n_labels))
+    for epoch_data in data:
+        for ii in range(n_labels):
+            for jj in range(n_labels):
+                # Get timeseries for each pair
+                x, y = epoch_data[ii], epoch_data[jj]
+                x_mag = np.abs(x)
+                x_conj_scaled = x.conj()
+                x_conj_scaled /= x_mag
+                # Calculate orthogonalization
+                y_orth_x = (y * x_conj_scaled).imag
+                y_orth_x_mag = np.abs(y_orth_x)
+                # Estimate correlation
+                corr[ii, jj] += np.abs(np.corrcoef(x_mag, y_orth_x_mag)[0, 1])
```

Comment:
@SherazKhan the actual critical change in the code below is from `data_mag_nomean[li]` which only used the `li`th label's data to correlate with all N labels `label_data_orth` series, and this code now uses `data_mag_nomean`, i.e., all `N` labels data to correlate with all N labels `label_data_orth` series.

It's probably easier to see if you look at the "simplified" version here. The old/`master` version of the code does the correlation with the `jj` original (magnitude) data that you could call `y_mag`:
```
                corr[ii, jj] += np.abs(np.corrcoef(np.abs(epoch_data[jj]), y_orth_x)[0, 1])
```
Instead of this correlation, with (magnitude) of the `ii`th data that is called `x_mag` here, which is now implemented in this PR, equivalently written as:
```
                corr[ii, jj] += np.abs(np.corrcoef(np.abs(epoch_data[ii]), y_orth_x)[0, 1])
```

Reply:
@larsoner indeed, this is more closer to the original Hipp version.